File: cn/java-spring-demo/src/main/java/com/itangcent/springboot/demo/common/model/ValidationDemoDto.java
package com.itangcent.springboot.demo.common.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

/**
 * 数据验证演示DTO
 * 用于展示JSR-303验证注解的使用
 * 包含基本的非空验证示例
 */
public class ValidationDemoDto {
    /**
     * ID字段
     * 不能为null
     */
    @NotNull
    private Long id;

    /**
     * 名称字段
     * 不能为空字符串或null
     */
    @NotBlank
    private String name;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
} 

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/common/model/ValidationDemoDto.kt
package com.itangcent.common.model

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

/**
 * Data Validation Demo DTO
 * Used to demonstrate JSR-303 validation annotations
 * Contains basic non-null validation examples
 */
data class ValidationDemoDto(
    /**
     * ID Field
     * Cannot be null
     */
    @NotNull
    var id: Long? = null,

    /**
     * Name Field
     * Cannot be blank string or null
     */
    @NotBlank
    var name: String? = null
)